### PR TITLE
Load the app/SP from the environment instead of automatically creating it

### PR DESCRIPTION
**Pull request by hawkowl**
### Which issue this PR addresses:

Part of ARO-5822

### What this PR does / why we need it:

Loads the application and SP secret from the env, instead of automatically creating it. This allows us to load them from a fixed pool in the e2e process, instead of creating new ones each time.

### Test plan for issue:

Manually tested creating a cluster locally with the steps updated in the docs

### Is there any documentation that needs to be updated for this PR?

Updated dev RP docs

### How do you know this will function as expected in production? 

Purely CI/E2E/dev changes -- more will need to happen for prod e2e

### CODE REVIEW

**Review by tsatam — APPROVED** [review]

**Review by mociarain — CHANGES_REQUESTED** [review]
Looking at it again I'm now very sure that removing the portal v1 line from the `.gitignore` will cause us issues

**Review by s-fairchild — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -30,7 +30,6 @@ gomock_reflect_*
 /e2e-report.xml
 /deploy/config.yaml
 **/*.swp
-/portal/v1/node_modules/
```
[the remark this answers is not in the document]
Comment (s-amann) [reply .gitignore:33]: because #3465 was merged

**Review by s-amann — APPROVED** [review]

File: .gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (mociarain) [reply .gitignore:33]: Now true ;)

**Review by mociarain — APPROVED** [review]

File: hack/e2e/run-rp-and-e2e.sh [hunk 2]
```diff
@@ -211,9 +211,31 @@ delete_e2e_cluster() {
...
     else
         go run ./hack/cluster delete
+        go run ./hack/cluster deleteApp
     fi
 }
 
+get_cluster_sp() {
+    echo "########## Downloading SP secrets ##########"
+
+    az keyvault secret download --vault-name=aro-e2e-principals \
```
Comment (tsatam) [inline hack/e2e/run-rp-and-e2e.sh:221]: Not blocking the PR, but we should add a TODO here for eventually grabbing a single SP among the entire pool

**Review by tsatam — APPROVED** [review]

**Pull request merged**